src/tutorials/boss_tutorial.md

```diff
@@ -0,0 +1,4330 @@
+
+3. Add property references for projectile and shoot position.
+
+    ```lua
+    local PROJECTILE = script:GetCustomProperty("Projectile")
+    local SHOOT_POSITION = script:GetCustomProperty("ShootPosition"):WaitForObject()
+    ```
+
+    Add the 2 lines above to the top of your script.
+
+4. Add projectile code to `ShootProjectile` function.
+
+    In an earlier section, you added the `ShootProjectile` function, which is an empty function. Add the code below in replace of the `ShootProjectile` function.
+
+    ```lua
+    local function ShootProjectile()
+        if isCharging then
+            return
+        end
+
+        isCharging = true
+
+        local startPos = SHOOT_POSITION:GetWorldPosition()
+
+        Events.BroadcastToAllPlayers("PlayChargeUpEffect")
+
+        Task.Wait(.5)
+
+        if not Object.IsValid(BOSS_GEO) then
+            return
+        end
+
+        local direction = BOSS_GEO:GetWorldRotation() * Vector3.FORWARD
+        local theProjectile = Projectile.Spawn(PROJECTILE, startPos, direction)
+
+        theProjectile.speed = 7000
+        theProjectile.shouldDieOnImpact = false
+        theProjectile.lifeSpan = 3.5
+        theProjectile.gravityScale = 0.1
+
+        isCharging = false
+        hasShot = true
+    end
+    ```
+
+    The `ShootProjectile` function will make sure there is a valid target to shoot at, if there is, spawn a projectile.
+
+    ```lua
+    Events.BroadcastToAllPlayers("PlayChargeUpEffect")
+    ```
+
+    The **Boss Geo** template you added to your **Hierarchy** comes with an effect script called **EffectsClient** that will play a charge up effect for all players to see. This is added to help players have time to react to the projectile being shot at them.
+
+    ```lua
+    local direction = BOSS_GEO:GetWorldRotation() * Vector3.FORWARD
+    ```
+
+    The projectile needs to know the direction to head towards. Because the boss rotates when tracking a target, then the direction also changes. You can get the forward direction of an object using `Vector3.FORWARD`, and multiplying the forward direction with the **Boss Geo** rotation will give us the direction the boss is facing.
+
+    ```lua
+    local theProjectile = Projectile.Spawn(PROJECTILE, startPos, direction)
+    ```
+
+    A projectile is spawned at the `startPos`, and will shoot in the direction the boss is facing.
+
+    ```lua
+    theProjectile.shouldDieOnImpact = false
+    ```
+
+    Because of the type of projectile effect being used, you do not want the projectile to die on impact, because the trail effect from the projectile would be instantly removed.
+
+### Create Boss Projectile Impact Code
+
+The projectile does not do any damage to the player, or the floor tiles. You need to add impact code to handle what happens to those impacted objects.
+
+1. Add `GetDamageable` function.
```

Comment:
In general I would use "make" or "create" for any function that they're typing themselves, and "add" for things where they are using previously created parts.